Reference/Common-Pitfalls/index.md

```diff
@@ -0,0 +1,466 @@
+# Common Pitfalls & Anti-Patterns
+
+_This section is ultra important! It will describe many common pitfalls that developers fall in to. 
+Some of the anti-patterns mentioned here can bring your site to a grinding halt, cause memory leaks, or just make your site unstable or perform poorly.
+Make sure you read this section - it might just save your site!_
+
+## Usage of Singletons and Statics
+
+Generally speaking if you are writing software these days you should be using Dependency Injection principles. 
+If you do this, you probably aren't using Singletons or Statics (and for the most part you shouldn't be!), 
+however since Umbraco itself is not built with an IoC container to use out of the box you may find yourself 
+using Umbraco's built in Singleton accessors like: 
+`ApplicationContext.Current` or `UmbracoContext.Current`. In most cases you shouldn't be using these Singleton accessors, 
+it makes your code very difficult to test but more importantly using Singletons and Statics in your code make it very hard 
+to manage, APIs become leaky and ultimately you'll end up with more problems than when you started.
+
+In all Umbraco base classes that you'll normally use, these object are already exposed as properties, so please use these instead!
+For example, all Razor views that Umbraco creates expose an `Umbraco` property which is the UmbracoContext, they expose an `ApplicationContext` 
+property which is Umbraco's `ApplicationContext`. The other base classes that expose all the instances you need are things like `SurfaceController`,
+`UmbracoApiController`, `UmbracoController`, `RenderMvcController`, `UmbracoUserControl`, `UmbracoPage`, `UmbracoHttpHandler`, and the list goes on...
+
+So next time you are using `ApplicationContext.Current` or `UmbracoContext.Current` think "Why am I doing this?", 
+"Is this already exposed as a property of the base class that I'm using?", "I'm using Dependency Injection, I should be injecting this instance into my class."
+
+
+## Static references to web request instances (such as `UmbracoHelper`)
```

Comment:
Ace @Shazwazza you talk about what not to do in the examples, but there is no follow up example of how it should be done, I know you mention it's available in the base classes, but perhaps pick an example base common class probably a SurfaceController & show how to access the membership helper in a simple code snippet.

So ie. don't do this but do this, however the technical explanation reads well & makes sense why you shouldn't do it that way.